src/workload/mod.rs

```diff
@@ -723,3 +724,35 @@ pub async fn reconnect(work_sender: Sender<ClientWorkItem>, config: Config) -> R
 
     Ok(())
 }
+
+#[derive(Clone)]
+pub struct Ratelimit {
+    start: u64,
+    end: u64,
+    step: u64,
+    current: u64,
+}
+
+impl Ratelimit {
+    pub fn new(start: Option<NonZeroU64>, end: Option<u64>, step: Option<u64>) -> Self {
```

Comment:
should this just load itself from the config instead?

Reply:
Good call, done.